### ISSUE TITLE
Upgrade manager fails if current version is already desired version

### ISSUE THREAD

**Issue by wrdls** [litmuschaos/litmus#3551]
**What happened**:

We deploy Litmus with the [Helm chart](https://github.com/litmuschaos/litmus-helm/tree/master/charts/litmus) with [Argo CD](https://argo-cd.readthedocs.io/en/stable/).

The Helm chart has a [post-upgrade-hook](https://github.com/litmuschaos/litmus-helm/blob/master/charts/litmus/templates/post-upgrade-hook.yaml) which runs the upgrade manager process.

Argo CD however runs a periodic sync process which triggers this hook every time which causes the sync to fail because of the below error.

```
{"level":"fatal","ts":1649855942.541615,"caller":"upgrade-agent/main.go:54","msg":"failed to create upgrade manager","error":"previous version and current version are same","stacktrace":"main.main\n\t/upgrade-agent/main.go:54\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:225"}
```

**What you expected to happen**:

The upgrade process exits successfully if the current version is already the desired version (i.e. no upgrade needs to happen).

**Where can this issue be corrected? (optional)**

https://github.com/litmuschaos/litmus/blob/4085bc09efd21f98751b96a2a21859c5b289beaa/litmus-portal/upgrade-agents/control-plane/versions/upgrade-manager.go#L43

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Comment by brnck** [comment]
Still a problem in **2.8.1**

```
❯ helm upgrade --install litmus litmus --repo https://litmuschaos.github.io/litmus-helm/ --version 2.8.1 --namespace litmus --create-namespace -f dev.yaml
Release "litmus" does not exist. Installing it now.
NAME: litmus
LAST DEPLOYED: Wed Apr 27 13:11:50 2022
NAMESPACE: litmus
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Thank you for installing litmus 😀

Your release is named litmus and it's installed to namespace: litmus.

Visit https://docs.litmuschaos.io to find more info.
❯ helm upgrade --install litmus litmus --repo https://litmuschaos.github.io/litmus-helm/ --version 2.8.1 --namespace litmus --create-namespace -f dev.yaml
Error: UPGRADE FAILED: post-upgrade hooks failed: job failed: BackoffLimitExceeded
```

**Comment by Deepak1100** [comment]
+1

**Comment by Jonsy13** [comment]
HI @wrdls @Deepak1100 @brnck  Thanks for pointing this out.
This issue was fixed in 2.9.0 release of ChaosCenter.
Closing this issue for now. Feel free to open if you still face same issue.
